lib/shared/bucketing-assembly-script/__tests__/bucketing/bucketing.test.ts

```diff
@@ -644,3 +649,57 @@ describe('Rollout Logic', () => {
         expect(doesUserPassRollout({ boundedHash: 0.9 })).toBeTruthy()
     })
 })
+
+describe('Client Data', () => {
+    it('uses client data to allow a user into a feature', () => {
+        const user = {
+            user_id: 'client-test',
+            customData: {
+                favouriteFood: 'pizza',
+            },
+            platformVersion: '1.1.2'
+        }
+        const clientData = {
+            'favouriteFood': 'NOT PIZZA!!',
+            favouriteDrink: 'coffee'
+        }
+
+        const c1 = generateBucketedConfig({ config, user })
+        expect(c1).toEqual(expect.objectContaining({
+            'featureVariationMap': {}
+        }))
+
+        setClientCustomDataJSON(clientData)
+
+        const expected = {
+            'featureVariationMap': {
+                '614ef6aa473928459060721a': '615357cf7e9ebdca58446ed0',
+                '614ef6aa475928459060721a': '615382338424cb11646d7667'
+            }
+        }
+        const c2 = generateBucketedConfig({ config, user })
+        expect(c2).toEqual(expect.objectContaining(expected))
+
+        setClientCustomDataJSON({
+            'favouriteFood': 'pizza',
+            favouriteDrink: 'coffee'
+        }
+        )
+
+        const user2 = {
+            user_id: 'hates-pizza',
+            customData: {
+                favouriteFood: 'NOT PIZZA!!',
+            },
+            platformVersion: '1.1.2'
+        }
+
+        const c3 = generateBucketedConfig({ config, user: user2 })
+        expect(c3).toEqual(expect.objectContaining({
+            'featureVariationMap': {}
+        }))
+
+        // cleanup
+        setClientCustomDataJSON({})
```

Comment:
I would be in favour of having an actual cleanup method that we could use in the `close` method

Reply:
Well this was just meant to clean up the test's side effects so it wouldn't interfere with other tests in this file.

I can see a need to clean this up when a client is closed, but it would be similar to cleaning up the config data (which we also don't do right now).
Maybe it's better to tackle all of that at once, in a single "clean up this client from the WASM" type operation like you were working on before?

Reply:
yeah I think that it would be fine to tackle this separately as part of a larger cleanup task